apps/hash-api/src/graphql/resolvers/knowledge/user/submit-early-access-form.ts

```diff
@@ -128,6 +137,23 @@ export const submitEarlyAccessFormResolver: ResolverFn<
     },
   );
 
+  // TODO: allow creating policies alongside entity creation
+  //   see https://linear.app/hash/issue/H-4622/allow-creating-policies-alongside-entity-creation
+  await createPolicy(context.graphApi, authentication, {
+    name: "prospective-user-view-entity",
+    principal: {
+      type: "actor",
+      actorType: "user",
+      id: user.accountId,
+    },
+    effect: "permit",
+    actions: ["viewEntity"],
+    resource: {
+      type: "entity",
+      id: extractEntityUuidFromEntityId(entity.entityId),
+    },
+  });
```

Comment:
This one we can handle similarly to the secret: allow any `accountId` matching the `webId` of an entity to see a Prospective User.

Reply:
Captured the additional conditions in H-4699: Support custom principal conditions on policies